PermanentOrg/web-app
# PER-8553: Install eslint and set up npm script

## Description
This PR installs eslint and makes the `npm run lint` script run the linter. This was accomplished through the [convert tslint to eslint angular schematic,](https://github.com/angular-eslint/angular-eslint) but set up to not clear out all tslint config yet. This incorporates eslint properly into Angular's config so a call to `ng lint` will also run the linter (and is actually what the npm script is pointing too). This also automatically changed the `tslint:disable` comments throughout the codebase (though we might just want to adjust those lines to not need the comments in the first place).

I also removed configuration for the "mdot-e2e" project in the `angular.json` file, since it doesn't seem to exist and all the config paths were also nonexistent. Since linting goes through Angular, and it tries to run linting on all projects, this means that calling `ng lint` tried to check these nonexistent end to end tests, which was causing a failure (obviously it still fails because of linting errors, but it doesn't fail because of an actual file path error).

Resolves PER-8553.

## Steps to Test
- Run `npm install`
- Run `npm run lint`
- You should see linter results for the codebase.
- Adjust the `.eslintrc.json` file and run again to verify that eslint, and not tslint, is running.
